angularSTEP/src/app/upload-recipe/upload-recipe.component.ts

```diff
@@ -151,77 +179,160 @@ export class UploadRecipeComponent{
+  async onSubmit() {
+    this.basicsFormGroup.controls.name.setValue(
+      this.autoCapitalizeName(this.basicsFormGroup.value.name)
+    );
+    this.basicsFormGroup.controls.description.setValue(
+      this.autoCapitalizeFirst(this.basicsFormGroup.value.description)
+    );
     this.formatArrays(this.ingredientsArray);
     this.formatArrays(this.toolsArray);
     this.formatInstructions(this.instructionsArray);
     this.extraFormGroup.controls.extraInfo.setValue(this.autoCapitalizeFirst(this.extraFormGroup.value.extraInfo));
 
-    this.db.collection('recipes')
-      .ref.withConverter(new RecipeConverter().recipeConverter)
-      .add(new Recipe(
-        this.basicsFormGroup.value.name,
-        '',
-        this.basicsFormGroup.value.difficulty,
-        this.basicsFormGroup.value.description,
-        this.ingredientsArray.value,
-        this.toolsArray.value,
-        [],
-        this.instructionsArray.value,
-        this.extraFormGroup.value.extraInfo ? this.extraFormGroup.value.extraInfo : '',
-        Date.now(),
-        [], 
-        [],
-    ));
-    this.router.navigate(['/confirm-upload']);
-  }
-
-  //image functions
-
-  onFileSubmit() {
-    this.fAuth.currentUser.then(user => {
+    this.fAuth.currentUser.then(async user => {
       if (user) {
-        //this.addImage(user.uid);
+        this.uploading = true;
+        //sets the urls from the image and waits for that to be done before proceeding
+        await this.setUrls();
+        this.db
+          .collection<Recipe>('recipes').ref.withConverter(new RecipeConverter().recipeConverter)
+          .add(new Recipe(
+            this.basicsFormGroup.value.name,
+            user.uid,
+            this.basicsFormGroup.value.difficulty,
+            this.basicsFormGroup.value.description,
+            this.ingredientsArray.value,
+            this.toolsArray.value,
+            this.imageUrls,
+            this.instructionsArray.value,
+            this.extraFormGroup.value.extraInfo
+              ? this.extraFormGroup.value
+              : '',
+            Date.now(),
+            [],
+            []
+          ))
+          .then(async recipeRef => {
+            //then it connects the recipe to the user signed in and adds it to their array
+            const userData = await this.afs
+              .collection('users')
+              .doc(user.uid)
+              .ref.withConverter(new Converter().userConverter)
+              .get();
+            if (userData && userData.data()) {
+              const tempUser: User = userData.data()!;
+              tempUser.recipes.push(recipeRef.id);
+              this.afs
+                .collection('users')
+                .doc(user.uid)
+                .ref.withConverter(new Converter().userConverter)
+                .update({recipes: tempUser.recipes})
+                .then(() => {
+                  this.uploading = false;
+                  console.log("why????");
```

Comment:
Might want to remove this `console.log("why????");`

Reply:
Oops! Thanks